### ISSUE TITLE
MacOS Idle detection is broken overnight

### ISSUE THREAD

**Issue by ytsvetukhin** [toggl/toggldesktop#1884]
![monosnap 2016-04-26 13-20-50](https://cloud.githubusercontent.com/assets/66028/14815147/59ae7db6-0bb2-11e6-9acf-2f2abb2efb09.png)

this is weekly situation with me when I forget to stop timer "yesterday" and open my mac "today". 
Idle detection is broken and can detect only last 5-10mins and all previous idle time is merged with my last activity(17:09:57 on screenshot) 

Usually, it happens while I open my mac and connect it to 2nd screen, mouse etc and it goes to sleep again

**Comment by IndrekV** [comment]
This is probably happening because a new idle time was triggered. App detects that idle time ends and shows the notification window and then starts a new one. And the one in the screenshot is the seconds one that has only 7 minutes tracked.

**Comment by ytsvetukhin** [comment]
yes, I think that you are right it's happening on 2nd idle, but this is bad behavior for me as a user.
I don't expect that first idle will be automatically merged with my last time entry.

1st idle should 
either remain if possible 
or autocreate new time entry with name "idle since 01:11pm"

**Comment by wmertens** [comment]
I think this is a duplicate of #1368 

What I believe is happening is that the Apple PowerNap feature turns on the computer to download email etc, and for some reason this registers as not being idle. This used to work, last year or so, probably around the time of that other issue.

**Comment by CelineDesign** [comment]
Same issue here with Apple PowerNap disable.

**Comment by NghiaTranUIT** [comment]
I found this valuable reference here: https://stackoverflow.com/questions/27377686/firefox-extension-addon-detect-workspace-locked

Basically, it mimic how chromium source handle Idle or lock/unlock screen on Windows and macOS. So, We should tamper it carefully ⛑
